### PR TITLE
[WIP] renaming for batch action logic

### PR DESCRIPTION
**Pull request by OskarStark**
I am targeting this branch, because this should be result, if you all except I will deprecate the usage.

## Changelog

```markdown
### Changed
- naming strategy for batch actions
```

## To do

- [ ] Update the tests
- [ ] Update the documentation
- [ ] Add an upgrade note

## Subject

I like the wording `mergeBatchAction`, which sounds better, instead of `batchActionMerge`.

WDYT @sonata-project/contributors

### CODE REVIEW

**Comment by OskarStark** [comment]
Can someone please "Request changes", that merging gets blocked? Thanks

File: Controller/CRUDController.php [hunk 1]
```diff
@@ -437,8 +437,8 @@ public function batchAction()
             ), null);
         }
 
-        // execute the action, batchActionXxxxx
-        $finalAction = sprintf('batchAction%s', $camelizedAction);
+        // execute the action, xxxxxBatchAction
+        $finalAction = sprintf('%sBatchAction', $camelizedAction);
```
Comment (core23) [inline Controller/CRUDController.php]: Can you find a BC way for this? 

Maybe something like:

```php
$finalAction array_filter(array(
    sprintf('%sBatchAction', $camelizedAction), 
    sprintf('batchAction%s', $camelizedAction)
), 'is_callable');

```

```

Comment (OskarStark) [inline Controller/CRUDController.php]: ofc, I think I can, but do you like the idea of the PR?

File: Controller/CRUDController.php [hunk 2]
```diff
@@ -125,9 +125,9 @@ public function listAction()
      *
      * @throws AccessDeniedException If access is not granted
      */
-    public function batchActionDelete(ProxyQueryInterface $query)
+    public function deleteBatchAction(ProxyQueryInterface $query)
```
Comment (Soullivaneuh) [inline Controller/CRUDController.php]: Renaming method is BC break for PHP class extension. I don't how this could be BC.

Comment (core23) [reply Controller/CRUDController.php]: Maybe you could write a wrapper method @OskarStark

File: Controller/CRUDController.php [hunk 3]
```diff
@@ -125,9 +125,9 @@ public function listAction()
...
     {
-        $this->admin->checkAccess('batchDelete');
+        $this->admin->checkAccess('deleteBatch');
```
Comment (jlamur) [inline Controller/CRUDController.php]: Is it going to work?

File: Controller/CRUDController.php (hunk 2 above)
Comment (OskarStark) [reply Controller/CRUDController.php]: ofc I can, and I know, that this is a BC break, but I don't have time to work on this PR

**Pull request closed without merging**